### PR TITLE
[IOPID-731] : bff unit test

### PR DESCRIPTION
**Pull request by pelliccm**
#### List of Changes

Unit tests for exchange-jwt-validation and user-eligibility middlewares. 
Files required to make them work (also included in [IOPID-729 PR](https://github.com/pagopa/io-web-profile-backend/pull/16))

#### Motivation and Context

Testing middlewares, functions and higher coverage

#### How Has This Been Tested?

Unit tests

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,8 +84,14 @@ The following table contains the required ENV variables that the function requir
...
+| EXCHANGE_JWT_PUB_KEY            | Exchange JWT Public Key (token validation)          | string|
+| EXCHANGE_JWT_PRIVATE_KEY        | Exchange JWT Private Key (token validation)         | string|
+| EXCHANGE_JWT_ISSUER             | Exchange JWT ISSUER                                 | string|
+| EXCHANGE_JWT_TTL                | Exchange JWT TTL in seconds                         | string|
 | HUB_SPID_LOGIN_JWT_PUB_KEY      | Hub Spid Login JWT Public Key (token validation)    | string|
 | HUB_SPID_LOGIN_JWT_ISSUER       | Hub Spid Login ISSUER                               | string|
 | HUB_SPID_LOGIN_CLIENT_BASE_URL  | Hub Spid Login baseUrl                              | string|
 | FAST_LOGIN_API_KEY              | Fast Login Api Key                                  | string|
 | FAST_LOGIN_CLIENT_BASE_URL      | Fast Login baseUrl                                  | string|
+| FUNCTIONS_APP_SUBSCRIPTION_KEY  | Io Functions App Api Key                            | string|
```
Comment (gquadrati) [inline README.md:96]: This should be a function key `FUNCTIONS_APP_API_KEY`. Change it also in `config.ts`

File: Info/handler.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ type InfoHandler = () => Promise<
 
 type HealthChecker = (
   config: unknown
-) => healthcheck.HealthCheck<"AzureStorage" | "Config" | "AzureCosmosDB", true>;
+) => healthcheck.HealthCheck<"AzureStorage" | "Config", true>;
```
Comment (gquadrati) [inline Info/handler.ts:26]: Unit tests are failing because of this change.

File: utils/featureFlags/__test__/featureFlags.test.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  it("should succeed in passing the user as eligible when NONE", async () => {
+    const fiscalCode = "ISPXNB32R82Y766D";
+    const result = getIsUserEligibleForNewFeature(
+      (fc: string) => betaTester.includes(fc),
+      (fc: string) => canaryTester.includes(fc),
+      FeatureFlagEnum.NONE
+    )(fiscalCode);
+
+    expect(result).toBe(false);
+  });
```
Comment (gquadrati) [inline utils/featureFlags/__test__/featureFlags.test.ts]: We can copy tests from https://github.com/pagopa/io-functions-services/blob/master/utils/__tests__/featureFlag.test.ts

File: SessionState/__test__/handler.test.ts [hunk 4]
```diff
@@ -132,5 +132,30 @@ describe("SessionState", () => {
       kind: "IResponseErrorGatewayTimeout"
     });
   });
+
+  test(`GIVEN a valid user decoded from JWT
+    WHEN the client returns an error
+      THEN the response should be 500`, async () => {
```
Comment (gquadrati) [inline SessionState/__test__/handler.test.ts:138]: As a suggestion, in future consider the option of making similar tests parametric using `it.each`.

File: SessionState/__test__/handler.test.ts [hunk 5]
```diff
@@ -132,5 +132,30 @@ describe("SessionState", () => {
...
+    const res = await handler(aValidUser);
+
+    expect(sessionStateMock).toHaveBeenCalledTimes(1);
+    expect(sessionStateMock).toHaveBeenCalledWith({
+      body: {
+        fiscal_code: aValidUser.fiscal_number
+      }
+    });
+    expect(res).toMatchObject({
+      kind: "IResponseErrorInternal"
```
Comment (gquadrati) [inline SessionState/__test__/handler.test.ts:160]: We could check `detail` too.

**Review by gquadrati — CHANGES_REQUESTED** [review]
LGTM, just a few small changes

File: README.md (hunk 1 above)
Comment (pelliccm) [reply README.md:96]: I corrected this in [another PR](https://github.com/pagopa/io-web-profile-backend/pull/16/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)

**Review by gquadrati — APPROVED** [review]
LGTM, thanks!

**Pull request merged**